root_draft_metadata.tf

```diff
@@ -46,9 +47,212 @@ module "draft_metadata_api_gateway" {
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "api_invoke_policy" {
+  name = "TDRAPIInvokePolicy${title(local.environment)}"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = "states:InvokeHTTPEndpoint"
+        Resource = module.draft_metadata_checks.step_function_arn
+        Condition = {
+          StringEquals = {
+            "states:HTTPMethod" = "POST"
+          }
+          StringLike = {
+            "states:HTTPEndpoint" = "${module.consignment_api.api_url}/*"
+          }
+        }
+      },
+      {
+        Effect   = "Allow"
+        Action   = "events:RetrieveConnectionCredentials"
+        Resource = aws_cloudwatch_event_connection.consignment_api_connection.arn
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:DescribeSecret"
+        ]
+        Resource = "arn:aws:secretsmanager:${local.region}:${var.tdr_account_number}:secret:events!connection/*"
+      }
+    ]
+  })
+}
+
+module "draft_metadata_checks" {
+  source             = "./da-terraform-modules/sfn"
+  step_function_name = "TDRMetadataChecks${title(local.environment)}"
+  step_function_definition = jsonencode(
+    {
+      "Comment" : "Run antivirus checks on metadata, update DB if positive, else trigger metadata validation",
+      "StartAt" : "RunAntivirusLambda",
+      "States" : {
+        "RunAntivirusLambda" : {
+          "Type" : "Task",
+          "Resource" : module.yara_av_v2.lambda_arn
+          "Parameters" : {
+            "consignmentId.$" : "$.consignmentId",
+            "fileId" : "draft-metadata.csv",
+            "scanType" : "metadata"
+          },
+          "ResultPath" : "$.output",
+          "Next" : "CheckAntivirusResults"
+        },
+        "CheckAntivirusResults" : {
+          "Type" : "Choice",
+          "Choices" : [
+            {
+              "Variable" : "$.output.antivirus.result",
+              "StringEquals" : "",
+              "Next" : "RunValidateMetadataLambda"
+            },
+            {
+              "Not" : {
+                "Variable" : "$.output.antivirus.result",
+                "StringEquals" : ""
+              },
+              "Next" : "PrepareVirusDetectedQueryParams"
+            }
+          ],
+          "Default" : "RunValidateMetadataLambda"
+        },
+        "PrepareVirusDetectedQueryParams" : {
+          "Type" : "Pass",
+          "ResultPath" : "$.statusUpdate",
+          "Parameters" : {
+            "query" : "mutation updateConsignmentStatus($updateConsignmentStatusInput: ConsignmentStatusInput!) { updateConsignmentStatus(updateConsignmentStatusInput: $updateConsignmentStatusInput) }",
+            "variables" : {
+              "updateConsignmentStatusInput" : {
+                "consignmentId.$" : "$.consignmentId",
+                "statusType" : "DraftMetadata",
+                "statusValue" : "Failed"
```

Comment:
I would have this as `CompletedWithIssues` The `Failed` status means that there is an actual failure with the code, ie the step function fails to run, the Lambda crashes etc. Here the code executes correctly and is doing the right thing, so it completed but found a problem with the input.

There is needs to be a re-appraisal of the statuses for draft metadata and more generally though.